pkg/apis/networking/v1/types.go

```diff
@@ -174,3 +175,144 @@ type NetworkList struct {
+	// If network in remote cluster is all underlay, it is nil
+	TunnelNetID *uint32 `json:"tunnelNetID,omitempty"`
+}
+
+// ClusterCondition describes current state of a cluster.
+type ClusterCondition struct {
+	// Type of cluster condition, Ready or Offline.
+	Type ClusterConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status apiv1.ConditionStatus `json:"status"`
+	// Last time the condition was checked.
+	LastProbeTime metav1.Time `json:"lastProbeTime"`
+	// Last time the condition transit from one status to another.
+	// +optional
+	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty"`
+	// (brief) reason for the condition's last transition.
+	// +optional
+	Reason *string `json:"reason,omitempty"`
+	// Human readable message indicating details about last transition.
+	// +optional
+	Message *string `json:"message,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RemoteClusterList is a list of RemoteCluster resources
+type RemoteClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []RemoteCluster `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// RemoteSubnet is a specification for a RemoteSubnet resource
+type RemoteSubnet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RemoteSubnetSpec   `json:"spec"`
+	Status RemoteSubnetStatus `json:"status"`
+}
+
+// RemoteSubnetSpec is the spec for a RemoteSubnet resource
+type RemoteSubnetSpec struct {
+	Range       AddressRange `json:"range"`
+	Type        NetworkType  `json:"type"`
+	ClusterName string       `json:"clusterName"`
+}
+
+// RemoteSubnetStatus is the status for a RemoteSubnet resource
+type RemoteSubnetStatus struct {
+	LastModifyTime metav1.Time `json:"lastModifyTime"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RemoteSubnetList is a list of RemoteSubnetList resources
+type RemoteSubnetList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []RemoteSubnet `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// RemoteVtep is a specification for a RemoteVtep resource
+type RemoteVtep struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RemoteVtepSpec   `json:"spec"`
+	Status RemoteVtepStatus `json:"status"`
+}
+
+// RemoteVtepSpec is the spec for a RemoteVtep resource
+type RemoteVtepSpec struct {
+	ClusterName string   `json:"clusterName"`
+	NodeName    string   `json:"nodeName"`
+	VtepIP      string   `json:"vtepIP"`
+	VtepMAC     string   `json:"vtepMAC"`
+	PodIPList   []string `json:"podIPList"`
```

Comment:
rename PodIPList to IPList

Reply:
Is IPList rather ambiguous?

Reply:
Maybe EndpointIPList is okay, ip addresses for vxlan endpoints.